File: Android/BarkCloud.Android/app/src/main/java/com/barkfluff/BarkCloud/ui/login/LoginUiState.kt
package com.barkfluff.BarkCloud.ui.login

data class LoginUiState(
    val login: String = "",
    val password: String = "",
    val passwordVisible: Boolean = false,
    val otp: String = "",
    val otpRequired: Boolean = false,
    val isLoading: Boolean = false,
    val credentialsError: String? = null,
    val snackbarMessage: String? = null,
) {
    val canSubmit: Boolean
        get() = !isLoading &&
            login.isNotBlank() &&
            password.isNotBlank() &&
            (!otpRequired || otp.length == OTP_LENGTH)

    companion object {
        const val OTP_LENGTH = 6
    }
}

File: Ios/BarkCloud/BarkCloud/Features/Login/LoginUiState.swift
import Foundation

struct LoginUiState: Equatable {
    static let otpLength = 6

    var login: String = ""
    var password: String = ""
    var passwordVisible: Bool = false
    var otp: String = ""
    var otpRequired: Bool = false
    var isLoading: Bool = false
    var credentialsError: String? = nil
    var snackbarMessage: String? = nil

    var canSubmit: Bool {
        if isLoading { return false }
        if otpRequired {
            return otp.count == LoginUiState.otpLength
        }
        return !login.trimmingCharacters(in: .whitespaces).isEmpty && !password.isEmpty
    }
}
